src/inc/corprof.idl

```diff
@@ -2408,6 +2412,31 @@ interface ICorProfilerCallback8 : ICorProfilerCallback7
         [in] BOOL       fIsSafeToBlock);
 }
 
+[
+    object,
+    uuid(27583EC3-C8F5-482F-8052-194B8CE4705A),
+    pointer_default(unique),
+    local
+]
+interface ICorProfilerCallback9 : ICorProfilerCallback8
+{
+    // This event is triggered whenever a dynamic method is garbage collected
+    // and subsequently unloaded.
+
+    HRESULT DynamicMethodUnloaded([in] FunctionID functionId);
+
+    HRESULT EventPipeEventDelivered(
+        [in] REFGUID eventGuid,
```

Comment:
as a proper COM interface we can't mutate it after it ships, and there is reasonable risk that whatever is commited here will ship as-is due to the rising bug-bar for changes. I don't want to ship an API that we know will be immediately deprecated.